src/EditorFeatures/CSharpTest/EditAndContinue/StatementEditingTests.cs

```diff
@@ -8900,6 +8909,253 @@ public void DeclarationPattern_Reorder()
                 "Reorder [j]@32 -> @16");
         }
 
+        [Fact]
+        public void VarPattern_Update()
+        {
+            var src1 = @"
+if (!(o is (var x, var y))) return;
+if (!(o4 is (string a, var (b, c)))) return;
+if (!(o2 is var (e, f, g))) return;
+if (!(o3 is var (k, l, m))) return;
+";
+
+            var src2 = @"
+if (!(o is (int x, int y1))) return;
+if (!(o1 is (var a, (var b, string c1)))) return;
+if (!(o7 is var (g, e, f))) return;
+if (!(o3 is (string k, int l2, int m))) return;
+";
+
+            var edits = GetMethodEdits(src1, src2, CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Preview));
+
+            edits.VerifyEdits(
+                "Update [if (!(o is (var x, var y))) return;]@4 -> [if (!(o is (int x, int y1))) return;]@4",
+                "Update [if (!(o4 is (string a, var (b, c)))) return;]@41 -> [if (!(o1 is (var a, (var b, string c1)))) return;]@42",
+                "Update [if (!(o2 is var (e, f, g))) return;]@87 -> [if (!(o7 is var (g, e, f))) return;]@93",
+                "Reorder [g]@110 -> @110",
+                "Update [if (!(o3 is var (k, l, m))) return;]@124 -> [if (!(o3 is (string k, int l2, int m))) return;]@130",
+                "Update [y]@27 -> [y1]@27",
+                "Update [c]@72 -> [c1]@77",
+                "Update [l]@144 -> [l2]@157");
+        }
+
+        [Fact]
+        public void PositionalPattern_Update()
```

Comment:
>PositionalPattern_Update [](start = 20, length = 24)

This is too complicated. I don't see if the behavior is actually correct or not.

Reply:
1. I split this into two tests. 
2. The reorder happens because the order of int parameters changed:
(0, var b, int c)
a. move int parameter from slot 2 (starting with 0) to slot 1.
(0, int c, var b)

b. rename c to b1
(0, int b1, var b)

c. rename b to c1
(0, int b1, var c1)

---
In reply to: [301337951](https://github.com/dotnet/roslyn/pull/37015#discussion_r301337951) [](ancestors = 301337951)